site/en/blog/immutable-document-domain/index.md

```diff
@@ -0,0 +1,197 @@
+but are different subdomains are considered "same-site but cross-origin".
+
+{% endAside %}
+
+Here's how it's used:
+
+Let's say a page on `https://parent.example.com` embeds an iframe page from
+`https://video.example.com`. These pages have the same eTLD+1 (`example.com`)
+with different subdomains. By setting both pages' `document.domain` to
+`'example.com'`, the two origins can be treated as if they are the same-origin
+by the browser.
+
+On `https://parent.example.com`:
+
+```js
+console.log(document.domain); // "parent.example.com"
+document.domain = 'example.com';
+console.log(document.domain); // "example.com"
+```
+
+On `https://video.example.com`:
+
+```js
+console.log(document.domain); // "video.example.com"
+document.domain = 'example.com';
+console.log(document.domain); // "example.com"
+```
+
+On `https://parent.example.com`, do a cross origin DOM manipulation against
+`https://video.example.com`.
+
+Websites set `document.domain` to make it possible for same-site documents to
+communicate more easily. Because this change [relaxes the same-origin
+policy](https://html.spec.whatwg.org/multipage/origin.html#relaxing-the-same-origin-restriction),
+the parent page will be able to access the iframe's document and traverse the
+DOM tree, and vice versa.
+
+At a glance, this is a convenient technique, but this actually introduces a
+security risk.
+
+### Security concerns with `document.domain`
+
+Security concerns have led to the spec for `document.domain` to [warn users to
+avoid
+usage](https://html.spec.whatwg.org/multipage/origin.html#relaxing-the-same-origin-restriction).
+The [current discussion with other browser
+vendors](https://github.com/w3ctag/design-reviews/issues/564) is moving toward
+the same direction.
+
+For example, if a hosting service provides different subdomains per user, an
+attacker can set `document.domain` to pretend as if they are the same-origin
+with another user's page. Further, an attacker can host a website under a shared
+hosting service, which serves through the same IP address with different port
+numbers, the attacker can pretend to be on the same-site-but-same-origin as
+yours. This is possible because `document.domain` ignores the port number part
+of the domain.
+
+To learn more about the security implications of setting `document.domain`, read
+["Document.domain" page on
+MDN](https://developer.mozilla.org/docs/Web/API/Document/domain#setter).
+
+Chrome will start displaying a warning in DevTools Console as soon as
+`document.domain` is set starting Chrome 98. Chrome is planning to  make
+`document.domain` immutable starting Chrome 101.
+
+## Alternative cross-domain communication
+
+### Use `postMessage()` or Channel Messaging API
+
+In most use cases, `document.domain` can be replaced by a cross-origin
+[`postMessage()`
+](https://developer.mozilla.org/docs/Web/API/Window/postMessage) or [Channel
+Messaging
+API](https://developer.mozilla.org/docs/Web/API/Channel_Messaging_API).
```

Comment:
```suggestion
In most use cases, cross-origin
[`postMessage()`
](https://developer.mozilla.org/docs/Web/API/Window/postMessage) or [Channel
Messaging
API](https://developer.mozilla.org/docs/Web/API/Channel_Messaging_API) can replace `document.domain`.
```

Reply:
This is just a suggestion in line with the evergreen advice of avoiding passive voice whenever possible.